lib/ansible/plugins/action/template.py

```diff
@@ -62,6 +62,8 @@ def run(self, tmp=None, task_vars=None):
         dest   = self._task.args.get('dest', None)
         force  = boolean(self._task.args.get('force', True))
         state  = self._task.args.get('state', None)
+        decode = self._task.args.pop('decode', None)
+        encode = self._task.args.pop('encode', None)
```

Comment:
The default of to_text and to_bytes is utf-8 so we can use None as the default value here if we want.